### ISSUE TITLE
softmask backdrop ignored

### ISSUE THREAD

**Issue by THausherr** [mozilla/pdf.js#7891]
Link to PDF file (or attach file here):

[gs-bugzilla689309-reduced-bc1_RGB.pdf](https://github.com/mozilla/pdf.js/files/649453/gs-bugzilla689309-reduced-bc1_RGB.pdf)
[gs-bugzilla689309-reduced-bc0_RGB.pdf](https://github.com/mozilla/pdf.js/files/649454/gs-bugzilla689309-reduced-bc0_RGB.pdf)

Configuration:
- Web browser and its version: ff 50.0.2
- Operating system and its version: W7 64bit fully patched
- PDF.js version: most recent
- Is an extension: ?

Steps to reproduce the problem:
1. open demo viewer
2. open file gs-bugzilla689309-reduced-bc1_RGB.pdf

What is the expected behavior? (add screenshot)
![screenshot-adobe](https://cloud.githubusercontent.com/assets/6665575/21148639/2fb5d466-c159-11e6-9410-7dbfb5eff11f.jpg)

What went wrong? (add screenshot)
![screenshot-pdfjs](https://cloud.githubusercontent.com/assets/6665575/21148650/35e90808-c159-11e6-95cb-a60a91bff6e6.jpg)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
n/a

Additional comment:
The two files are identical except for the /bc value in the softmask. The softmask is smaller than the transparency group it is applied to. My understanding is that the /bc color is to be used instead of the softmask as an "influence" for the area that isn't covered by the softmask. I suspect that PDF.js just ignores /bc.

The fine black line doesn't belong there (although it happens to be useful as it shows the size of the softmask). The red line does belong there and shows the size of the transparency group.

The file is a reduced and modified version of one in https://bugs.ghostscript.com/show_bug.cgi?id=688601 to investigate a different problem than the one mentioned there.

**Comment by Snuffleupagus** [comment]
> I suspect that PDF.js just ignores /bc.

That's not correct, please see [src/core/evaluator.js#L443](https://github.com/mozilla/pdf.js/blob/00a006e466014439dd43813e5cd2d7ec86b79a9b/src/core/evaluator.js#L443) and e.g. [src/display/canvas.js#L701](https://github.com/mozilla/pdf.js/blob/00a006e466014439dd43813e5cd2d7ec86b79a9b/src/display/canvas.js#L701)

**Comment by brendandahl** [comment]
Some notes in case I don't get back to this for awhile:

The issue is when we compose the smask we only compose the region that the smask image overlaps the canvas, not the whole group/page. This can be fixed by 1) creating a full size page canvas  2) fill the canvas with the `bc` color 3) draw the smask 4) compose full smask and the page canvas. Unfortunately, if we do that we then have to compose two much larger images. It'd probably be better if we tracked bounding boxes for what we need to compose, but I think that will be quite a bit of work to implement.
